### PR TITLE
Matplotlib exporter scientific notation bugfix

### PR DESCRIPTION
**Pull request by penkod**
Fix setting labels and values in case of plots which use scientific notation.

Solves issue #1050

### CODE REVIEW

File: pyqtgraph/exporters/Matplotlib.py [hunk 1]
```diff
@@ -53,28 +53,46 @@ def cleanAxes(self, axl):
...
+                SIprefix_scale_left = self.item.axes['left']['item'].autoSIPrefixScale
+
             ax = fig.add_subplot(111, title=title)
             ax.clear()
             self.cleanAxes(ax)
             #ax.grid(True)
             for item in self.item.curves:
                 x, y = item.getData()
+
+                # pyqtgraph by default uses scientific notation for large
```
Comment (ixjlyons) [inline pyqtgraph/exporters/Matplotlib.py]: Just one copy of the comment at the top should be sufficient.

File: pyqtgraph/exporters/Matplotlib.py [hunk 2]
```diff
@@ -53,28 +53,46 @@ def cleanAxes(self, axl):
...
-            
+
             # get labels from the graphic item
             xlabel = self.item.axes['bottom']['item'].label.toPlainText()
             ylabel = self.item.axes['left']['item'].label.toPlainText()
             title = self.item.titleLabel.text
 
+            # pyqtgraph by default uses scientific notation for large
+            # values. Matplotlib values and label should reflect that too.
+            SIprefix_scale_default = 1.0 # default SI prefix scale value
```
Comment (ixjlyons) [inline pyqtgraph/exporters/Matplotlib.py]: I think a more straightforward implementation might loo something like:

```python
scale_x = scale_y = 1.0
if self.items.axes['bottom']['item'].autoSIPrefix: # note no need for == True here
    scale_x = self.item.axes['bottom']['item'].autoSIPrefixScale
if self.items.axes['left']['item'].autoSIPrefix:
    scale_y = self.item.axes['left']['item'].autoSIPrefixScale
```

Then simply apply scale_x/scale_y below without the conditionals.

**Comment by ixjlyons** [comment]
Hey @penkod, I can go ahead and build from your commits to implement the requested changes. If you prefer to do it yourself, let me know. Thanks for the contribution in any case.

**Comment by penkod** [comment]
Hi @ixjlyons. At the time being, I unable to do the necessary changes + the tests, so of course, if you wish you can implement the necessary changes yourself.

**Pull request closed without merging**